#MOLGENIS walltime=23:59:00 mem=8gb nodes=1 ppn=2

### variables to help adding to database (have to use weave)
#string sampleName
#string project
###
#string stage
#string checkStage

#string WORKDIR
#string projectDir
#string imputedVcf
#string bam

#string exonlist
#string readCountDir
#string readCountFile

echo "## "$(date)" Start $0"

getFile ${imputedVcf}
getFile ${bam}


${stage} BEDTools/${bedtoolsVersion}
${stage} SAMtools/${samtoolsVersion}
${stage} tabix/${tabixVersion}

mkdir -p ${readCountDir}


#I: $1 Location of QCd BAM files $2 Location of Condensed exonList file (Generated by GetExonInfo)
#O: Y.txt Gcc.txt file generated for RASQUAL Main.txt file with gene data C.txt counts per gene
#W: Make sure the formats are the same for exonlist and BAMS chr1/1..chrX/X
###########################################################################
# Gets counts per exon per sample prints them to files and pastes them together to generate  CountTable (EXONLIST + Reads per sample)
# This is the longest step of the script
echo Intersecting with bams

bedtools intersect -sorted -c -a $exonlist -b ${bam} | awk -F "\t" '{ print $15 }' $f > part$COUNTER


