#MOLGENIS walltime=23:59:00 mem=40gb nodes=1 ppn=2

### variables to help adding to database (have to use weave)
#string sampleName
#string project
###
#string stage
#string checkStage

#string WORKDIR
#string projectDir
#string bam

#string exonSAF
#string readCountDir
#string readCountFileGene
#string readCountFileExon

#string pathSubread
#string sampleName

echo "## "$(date)" Start $0"
getFile ${bam}

mkdir -p ${readCountDir}


#I: $1 Location of QCd BAM files $2 Location of Condensed exonList file (Generated by GetExonInfo)
#O: Y.txt Gcc.txt file generated for RASQUAL Main.txt file with gene data C.txt counts per gene
#W: Make sure the formats are the same for exonlist and BAMS chr1/1..chrX/X
###########################################################################
# Gets counts per exon per sample prints them to files and pastes them together to generate  CountTable (EXONLIST + Reads per sample)
# This is the longest step of the script
echo Intersecting with bams
# Reads per Gene
PATH=$PATH:${pathSubread}
featureCounts \
	-F SAF \ # Annotation type = SAF
	-O \ # Allow a read to span two features (Not multimappers!)
	-s 1 \ #stranded = yes
	-p \ # count fragment not reads (for paired-end data)
	-B \ # count only both ends mapped
	-a ${exonSAF} \ # input file
	-o $TMPDIR/${sampleName}.txt ${bam} # output file + Input bam
tail -n +3 $TMPDIR/${sampleName}.txt | LC_ALL=C sort -t$'\t' -k1,1 | cut -f7 > ${readCountFileGene}
## Per Exon
featureCounts \
	-F SAF \ # Annotation type = SAF
	-O \ # Allow a read to span two features (Not multimappers!)
	-f \ # Count features not metafeatures ( Exons not genes)
	-s 1 \ #stranded = yes
	-p \ # count fragment not reads (for paired-end data)
	-B \ # count only both ends mapped
	-a ${exonSAF} \ # input file
	-o $TMPDIR/${sampleName}.txt ${bam} # output file + Input bam
tail -n +3 $TMPDIR/${sampleName}.txt | cut -f7 > ${readCountFileExon}
################################
echo "## "$(date)" ##  $0 Done "
################################
